File: ruby/raindrops/raindrops.rb
class Raindrops

  RAINDROPS = { "Pling" => 3, "Plang" => 5, "Plong" => 7 }

  def self.convert(num)
    result = RAINDROPS.select { |_key, value| num % value == 0 }.keys.join
    result.empty? ? num.to_s : result
  end
end

File: csharp/raindrops/Raindrops.cs
public class Raindrops
{
  public static string Convert(int input)
  {
    string result = "";

    if (input % 3 == 0)
      result += "Pling";
    if (input % 5 == 0)
      result += "Plang";
    if (input % 7 == 0)
      result += "Plong";

    return string.IsNullOrEmpty(result) ? input.ToString() : result;
  }
}